identity/src/commonMain/kotlin/com/android/identity/asn1/ASN1BitString.kt

```diff
@@ -0,0 +1,92 @@
+package com.android.identity.asn1
+
+import kotlinx.io.bytestring.ByteStringBuilder
+
+class ASN1BitString(
+    val numUnusedBits: Int,
+    val value: ByteArray
+): ASN1PrimitiveValue(tag = TAG_NUMBER) {
+
+    constructor(booleanValues: BooleanArray):
+            this(
+                if (booleanValues.size == 0) 0 else (8 - (booleanValues.size % 8)),
```

Comment:
The number of unused bits should always be in the range 0-7. If the number of bits is a multiple of 8, this will set the number of unused bits to 8, and we may end up skipping a byte when decoding (or, at the very least, it's inconsistent with the standard).

Maybe: ((8 - (booleanValues.size % 8)) & 0x07)
Or, equivalently: ((8 - (booleanValues.size % 8)) % 8)
?